### PR TITLE
Zsh: enable dynamic modules

### PR DESCRIPTION
**Pull request by miqlas**
Dirty patch, but works ok with zpresto

### CODE REVIEW

File: app-shells/zsh/patches/zsh-5.4.1.patchset [hunk 1]
```diff
@@ -127,4 +125,95 @@ index d1b98ac..eba94d8 100644
...
+
+diff --git a/configure.ac b/configure.ac
+index bad78f9..36d90c5 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -695,7 +695,6 @@ AC_CHECK_HEADERS(sys/time.h sys/times.h sys/select.h termcap.h termio.h \
+ 		 ncurses/ncurses.h)
+ if test x$dynamic = xyes; then
+   AC_CHECK_HEADERS(dlfcn.h)
+-  AC_CHECK_HEADERS(dl.h)
```
Comment (korli) [inline app-shells/zsh/patches/zsh-5.4.1.patchset]: What's the problem here? It should normally check the header and not find it, and continue.

File: app-shells/zsh/patches/zsh-5.4.1.patchset [hunk 2]
```diff
@@ -127,4 +125,95 @@ index d1b98ac..eba94d8 100644
...
+ fi
+ 
+ dnl Some SCO systems cannot include both sys/time.h and sys/select.h
+@@ -867,7 +866,7 @@ if test `echo $host_os | sed 's/^\(unicos\).*/\1/'` = unicos; then
+ fi
+ 
+ if test "x$dynamic" = xyes; then
+-  AC_CHECK_LIB(dl, dlopen)
++  AC_CHECK_LIB(root, dlopen)
```
Comment (korli) [inline app-shells/zsh/patches/zsh-5.4.1.patchset]: It should be possible to check multiple libs.

File: app-shells/zsh/patches/zsh-5.4.1.patchset [hunk 3]
```diff
@@ -127,4 +125,95 @@ index d1b98ac..eba94d8 100644
...
+ fi
+ 
+ if test x$enable_cap = xyes; then
+@@ -1433,8 +1432,7 @@ else
+ fi
+ 
+ if test x$dynamic = xyes; then
+-  AC_CHECK_FUNCS(dlopen dlerror dlsym dlclose load loadquery loadbind unload \
+-		shl_load shl_unload shl_findsym)
++  AC_CHECK_FUNCS(dlopen dlerror dlsym dlclose)
```
Comment (korli) [inline app-shells/zsh/patches/zsh-5.4.1.patchset]: Maybe explain that only these four functions are needed.

File: app-shells/zsh/patches/zsh-5.4.1.patchset (hunk 1 above)
Comment (miqlas) [inline app-shells/zsh/patches/zsh-5.4.1.patchset]: Will fix.

File: app-shells/zsh/patches/zsh-5.4.1.patchset (hunk 2 above)
Comment (miqlas) [inline app-shells/zsh/patches/zsh-5.4.1.patchset]: Will fix.

File: app-shells/zsh/patches/zsh-5.4.1.patchset (hunk 3 above)
Comment (miqlas) [inline app-shells/zsh/patches/zsh-5.4.1.patchset]: It built just fine without the others. Not sure if it actually correct or not.

**Comment by miqlas** [comment]
@korli : the configure script somewhere discriminates Haiku, plenty checks for everything and i don't seems to find, where it goes in wrong way, so i just shortwired all the checks in my patch, but as i wrote : it is a dirty patch, not for production or for upstreaming.
I'll look into it later again, but i think i'm just way too weak to do it correctly so i hope, maybe somebody with more knowledge can jump in and patch it correctly.

**Comment by miqlas** [comment]
@korli, i just fixed my patchset, it works now without messy things. Let me publish it later.

**Comment by miqlas** [comment]
Zsh still makes headache for haikuporter if it porvides cmd:sh, so i removed it again.

**Comment by miqlas** [comment]
It is now ready for merge.

**Pull request merged**